vpype_cli/operations.py

```diff
@@ -677,6 +686,12 @@ def pagerotate(document: vp.Document, clockwise: bool):
         return document
     w, h = page_size
 
+    # check orientation constraint, and do nothing if target orientation
+    # won't match desired result
+    if (orientation == "portrait" and h > w) or (orientation == "landscape" and w > h):
```

Comment:
A test for `orientation is None` is needed here because of the change above.

Reply:
Is an explicit test needed?  If `orientation == None`, then `orientation == "portrait"` will evaluate False.  The code seems to work fine without an explicit test here.

Reply:
I'd be surprised if `mypy` agreed with that, but let's see :) In any case, I generally dislike such implicit tests for `None`. Also, arguably any other string input should be checked and trigger a `logging.warning()`, and that requires an explicit check for `None`.

Reply:
well apparently mypy is on your side :)

Reply:
mypy or not, I'm good either way.  I can add it if you'd like.

Guido's 'explicit is better than implicit' perhaps applies here.

Reply:
The more I think of it, we should have that warning for (non-`None`) garbage input.

Reply:
I mean, if the use says `pagerotate --orientation garbage`. This should trigger a warning.

Reply:
I do realize now that we can't rely on click for people using it via the API...

Reply:
Added a check to the most recent commit.